src/probnum/statespace/sde.py

```diff
@@ -199,12 +210,57 @@ def backward_rv(
 
     # Forward and backward implementation(s)
 
-    def _solve_mde_forward(self, rv, t, dt, _diffusion=1.0):
+    def _solve_mde_forward_classic(self, rv, t, dt, _diffusion=1.0):
         """Solve forward moment differential equations (MDEs)."""
-        mde, y0 = self._setup_vectorized_mde_forward(
+        dim = rv.mean.shape[0]
+        mde, y0 = self._setup_vectorized_mde_forward_classic(
             rv,
             _diffusion=_diffusion,
         )
+
+        sol, new_mean, new_cov = self._solve_mde_forward(mde, y0, t, dt, dim)
+
+        # Useful for backward transitions
+        # Aka continuous time smoothing.
+        sol_mean = lambda t: sol.sol(t)[:dim]
+        sol_cov = lambda t: sol.sol(t)[dim:].reshape((dim, dim))
+
+        return randvars.Normal(mean=new_mean, cov=new_cov), {
+            "sol": sol,
+            "sol_mean": sol_mean,
+            "sol_cov": sol_cov,
+        }
+
+    def _solve_mde_forward_sqrt(self, rv, t, dt, _diffusion=1.0):
+        """Solve forward moment differential equations (MDEs) using a square-root
+        implementation."""
+        dim = rv.mean.shape[0]
+        mde, y0 = self._setup_vectorized_mde_forward_sqrt(
+            rv,
+            _diffusion=_diffusion,
+        )
+
+        sol, new_mean, new_cov_sqrt = self._solve_mde_forward(mde, y0, t, dt, dim)
+        new_cov = new_cov_sqrt @ new_cov_sqrt.T
+
+        # Useful for backward transitions
+        # Aka continuous time smoothing.
+        sol_mean = lambda t: sol.sol(t)[:dim]
+        sol_cov_sqrt = lambda t: sol.sol(t)[dim:].reshape((dim, dim))
+        sol_cov = (
+            lambda t: sol.sol(t)[dim:].reshape((dim, dim))
+            @ sol.sol(t)[dim:].reshape((dim, dim)).T
+        )
+
+        return randvars.Normal(mean=new_mean, cov=new_cov), {
```

Comment:
I think one can also initialise the `cov_cholesky` field of `Normal` with `new_cov_sqrt` (after all, that was the point here, right?). Or am I misunderstanding?

Reply:
Yes, I totally missed this.